.github/scripts/maven.functions_tests.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -eu
+
+function find_script_dir() {
+  CURRENT=$PWD
+
+  DIR=$(dirname "$0")
+  cd "$DIR" || exit
+  TARGET_FILE=$(basename "$0")
+
+  while [ -L "$TARGET_FILE" ]
+  do
+      TARGET_FILE=$(readlink "$TARGET_FILE")
+      DIR=$(dirname "$TARGET_FILE")
+      cd "$DIR" || exit
+      TARGET_FILE=$(basename "$TARGET_FILE")
+  done
+
+  local SCRIPT_DIR=$(pwd -P)
+  cd "$CURRENT" || exit
+  echo "$SCRIPT_DIR"
+}
+
+SCRIPT_DIR=$(find_script_dir)
+
+. "$SCRIPT_DIR"/assert.sh/assert.sh
```

Comment:
> can we use `source` rather than `.` I feel makes it more clearer and easy to see Not fussed though

I don't know the technical difference, but `source` is probably more explicit.

I've just followed the existing pattern in `_tests.sh` so maybe we could change them all separately?

Reply:
Understood. les leave in that case. I prefer consistency any day

BTW...no diff between `source` vs `.` as far as BASH is concerned but `.` is POSIX-standard while `source` is more readable and supported by bash and other shells. See https://stackoverflow.com/questions/20094271/using-dot-or-source-while-calling-another-script-what-is-the-difference